crates/context_aware_config/src/api/dimension/types.rs

```diff
@@ -1,18 +1,72 @@
 use chrono::{DateTime, NaiveDateTime, Utc};
+use derive_more::{AsRef, Deref, DerefMut, Into};
+use regex::Regex;
 use serde::{Deserialize, Deserializer, Serialize};
 use serde_json::Value;
 
 use crate::db::models::Dimension;
 
 #[derive(Debug, Deserialize)]
 pub struct CreateReq {
-    pub dimension: String,
-    pub priority: i32,
+    pub dimension: DimensionName,
+    pub priority: Priority,
     pub schema: Value,
     #[serde(default, deserialize_with = "deserialize_option")]
     pub function_name: Option<Value>,
 }
 
+#[derive(Debug, Deserialize, AsRef, Deref, DerefMut, Into)]
+#[serde(try_from = "i32")]
+pub struct Priority(i32);
+impl Priority {
+    fn validate_data(priority_val: i32) -> Result<Self, String> {
+        if priority_val <= 0 {
+            return Err("Priority should be greater than 0".to_string());
+        } else {
+            Ok(Self(priority_val))
+        }
+    }
+}
+
+impl TryFrom<i32> for Priority {
+    type Error = String;
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        Ok(Priority::validate_data(value)?)
+    }
+}
+
+#[derive(Debug, Deserialize, AsRef, Deref, DerefMut, Into)]
+#[serde(try_from = "String")]
+pub struct DimensionName(String);
+impl DimensionName {
+    pub fn validate_data(name: String) -> Result<Self, String> {
+        let regex_str: &str = "^[a-zA-Z0-9-_]([a-zA-Z0-9-_.]{0,254}[a-zA-Z0-9-_])?$";
+
+        let regex = Regex::new(regex_str).map_err(|err| {
+            log::error!("error while validating dimensionName : {err}");
+            "Something went wrong".to_string()
+        })?;
+
+        if !regex.is_match(&name) {
+            return Err(format!(
+                "The key name {} is invalid, it should obey the regex {}. \
+                It can contain the following characters only [a-zA-Z0-9-_.] \
```

Comment:
we can have this also extracted as a const `[a-zA-Z0-9-_.]`, being used twice, can be easier to maintain later
we can probably have some trait implemented here
something like, we can have an enum for all regexs

```
const KEY_NAME_REGEX: &str = "^[a-zA-Z0-9-_]([a-zA-Z0-9-_.]{0,254}[a-zA-Z0-9-_])?$";
const KEY_NAME_REGEX_WORDS: &str = "It can contain the following characters only [a-zA-Z0-9-_.] \
                                                                          and it should not start or end with a '.' character.";

const FUNCTION_NAME_REGEX: &str = "^[a-zA-Z0-9-_]{1,64}$";
const FUNCTION_NAME_REGEX_WORDS: &str = "It can contain the following characters only [a-zA-Z0-9-_]";

pub enum Regex {
   KeyName,
   FunctionName
}

impl Regex {
   fn get_error_message(&self) -> String {
      match self {
         Self::KeyName => KEY_NAME_REGEX_WORDS,
         Self::FunctionName => FUNCTION_NAME_REGEX_WORDS,
      }.to_string()
   }
}

impl Display for Regex {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        let regex = match self {
            Self::KeyName => KEY_NAME_REGEX,
            Self::FunctionName => FUNCTION_NAME_REGEX,
        }
        .to_string();
        write!(f, "{regex}")
    }
}
```

we can keep extending it further on addition of new other variants of the enum, it will help keep all the necessary things tight and mandatory

`TypeName` would also be a variant of it, not adding it in the code snippet, you can add this and others also